src/searchers/search_utils.cpp

```diff
@@ -69,3 +69,26 @@ void repetitionPenaltyLogitsProcess(float penalty, float *logits, int sampleOffs
         }
     }
 }
+
+void stopWordsCheck(std::vector<int> &nextTokenIds, std::vector<std::vector<int>> &stopWordsList,
+        std::vector<std::vector<int>> &stopWordsIndex, std::vector<int> &doneBatch) {
+    for (int batchId = 0; batchId < nextTokenIds.size(); batchId++) {
```

Comment:
@Duyi-Wang Do you need openmp for large batch size?

Reply:
Add TODO. We need more data and analysis to confirm the benefits of parallel.